src/app/shared/components/addscenario/addscenario.component.ts

```diff
@@ -168,9 +170,30 @@ export class AddScenarioModal implements OnInit, OnDestroy {
         });
     }
 
+    public equationCheck(check){
+        if (check == true) {
+            this.skipCheck = true;
+            this.newScenForm.get('regressionRegions.regressions.expected.value').disable();            
+            this.newScenForm.patchValue({ regressionRegions: { regressions: { expected: { value: null}}}});
+            const parmControl = <FormArray>this.newScenForm.get('regressionRegions.parameters');
+            for (let i = parmControl.length-1; i >= 0; i--) {
+                this.newScenForm.get('regressionRegions.parameters.'+i+'.value').disable();
+                parmControl.controls[i].get('value').setValue(null);
+            }
```

Comment:
So you can actually set this in the html, I think it's like `[disabled]="skipCheck"`, then this whole function can be removed and you can just change the skipcheck in the html instead.

Reply:
I tried to get this all to work in the html but I wasn't able too. I researched it a little and it seems you are unable use [disabled] within Reactive Forms, and instead are suppose to use the disable() method from the FormControl, but maybe I am misunderstanding?

Reply:
Oh, interesting. I thought we were using it somewhere in the app. Maybe we were using a disabled class instead? `[class.disabled]="skipCheck"`

Reply:
Or it could be that those aren't working/aren't in forms?

Reply:
We use [disabled] for buttons so I think those works because they are not in the forms, maybe.